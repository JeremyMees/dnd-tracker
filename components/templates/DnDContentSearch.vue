<script setup lang="ts">
const sortBy = ref<Open5eSortBy>('name')
const type = ref<Open5eType>('spells')
const page = ref<number>(0)
const limit = ref<number>(20)
const search = ref<string>('')
const debouncedSearch = refDebounced(search, 500, { maxWait: 1000 })
const cr = ref<string>()

const { data, status } = useOpen5eListing(computed(() => ({
  type: type.value,
  filters: {
    page: page.value,
    ...(debouncedSearch.value && { search: debouncedSearch.value }),
    ...(cr.value && { cr: cr.value }),
    ...(sortBy.value && { ordering: sortBy.value }),
  },
})))

watch(search, () => page.value = 0)

watch(type, () => {
  search.value = ''
  page.value = 0
  sortBy.value = 'name'
})
</script>

<template>
  <div class="flex flex-col gap-4 max-h-full">
    <Card
      color="secondary"
      class="flex flex-wrap items-center gap-4 w-full dnd-container"
    >
      <FormKit
        v-model="search"
        type="search"
        :label="$t('components.inputs.nameLabel')"
        outer-class="$reset !pb-0 flex-1"
      />
      <FormKit
        v-model="type"
        type="select"
        :label="$t('components.inputs.typeLabel')"
        :options="[
          { value: 'spells', label: $t('general.spell', 2) },
          { value: 'conditions', label: $t('general.condition', 2) },
          { value: 'magicitems', label: $t('general.magicItem', 2) },
          { value: 'weapons', label: $t('general.weapon', 2) },
          { value: 'armor', label: $t('general.armor') },
          { value: 'sections', label: $t('general.section', 2) },
        ]"
        outer-class="$reset !pb-0 flex-1"
        @input="search = ''"
      />
    </Card>

    <MasonryGrid
      v-if="status === 'pending'"
      v-slot="{ column }"
      :data="Array.from({ length: 30 }, () => ({}))"
    >
      <SkeletonContentCard
        v-for="(_item, i) in column"
        :key="i"
      />
    </MasonryGrid>
    <MasonryGrid
      v-else-if="data?.items?.length"
      v-slot="{ column }"
      :data="data.items"
    >
      <ContentCard
        v-for="(hit, j) in column"
        :id="j === 0 ? 'el' : ''"
        :key="hit.slug"
        :type="type"
        :hit="hit"
      />
    </MasonryGrid>

    <Pagination
      v-if="data?.pages > 1 && status !== 'pending' && data?.items?.length"
      v-model:page="page"
      :pages="data.pages"
      :per-page="limit"
      styled
      class="mx-auto"
      @paginate="scrollToId('el')"
    />
    <p
      v-if="status === 'error'"
      class="text-center max-w-prose mx-auto text-muted-foreground"
    >
      {{ $t('components.dndContentSearch.error') }}
    </p>
    <p
      v-if="status !== 'pending' && !data?.items?.length && search !== ''"
      class="text-center max-w-prose mx-auto text-muted-foreground"
    >
      {{ $t('components.dndContentSearch.notFound') }}
    </p>
  </div>
</template>
