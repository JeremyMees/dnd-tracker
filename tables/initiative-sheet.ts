import { createColumnHelper, type InitialTableState, type Row } from '@tanstack/vue-table'
import { actionsTable, expandButton, iconButton } from './generate-functions'

import {
  InitiativeTableRowName,
  InitiativeTableRowHealth,
  InitiativeTableRowActions,
  InitiativeTableRowConditions,
  InitiativeTableRowNotes,
  InitiativeTableRowInit,
  InitiativeTableRowAc,
  InitiativeTableRowDeathSaves,
  InitiativeTableRowConcentration,
  InitiativeTableRowModify,
} from '#components'

const columnHelper = createColumnHelper<InitiativeSheetRow>()

interface ColumnOptions {
  sheet: ComputedRef<InitiativeSheet | undefined>
  update: (payload: Omit<Partial<InitiativeSheet>, NotUpdatable>) => Promise<void>
  openQuickInitModal: () => void
}

export function generateColumns({ sheet, update, openQuickInitModal }: ColumnOptions) {
  const { t } = useI18n()

  return [
    columnHelper.display({
      enableGlobalFilter: false,
      id: 'index',
      cell: ({ row }) => h('div', { class: 'flex items-center gap-x-2 max-w-20' }, [
        h('span', { class: 'text-muted-foreground ml-1' }, row.index + 1),
        expandButton({
          content: t(`actions.${row.getIsExpanded() ? 'hide' : 'show'}`),
          expanded: row.getIsExpanded(),
          cb: () => row.toggleExpanded(),
        }),
      ]),
    }),
    columnHelper.accessor('name', {
      header: t('components.encounterTable.headers.name'),
      cell: ({ row }) => h(InitiativeTableRowName, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('initiative', {
      header: () => h('button', { class: 'flex items-center gap-x-2' }, [
        t('components.encounterTable.headers.init'),
        iconButton({
          icon: 'tabler:bolt-filled',
          content: t('components.encounterTable.quick'),
          color: 'warning',
          cb: () => openQuickInitModal(),
          disabled: !sheet.value?.rows.length,
          iconColor: 'text-warning',
        }),
      ]),
      cell: ({ row }) => h(InitiativeTableRowInit, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('health', {
      header: t('components.encounterTable.headers.health'),
      cell: ({ row }) => h(InitiativeTableRowHealth, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('ac', {
      header: t('components.encounterTable.headers.ac'),
      cell: ({ row }) => h(InitiativeTableRowAc, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.display({
      id: 'manage',
      header: t('components.encounterTable.headers.manage'),
      cell: ({ row }) => h(InitiativeTableRowActions, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('conditions', {
      header: t('components.encounterTable.headers.conditions'),
      cell: ({ row }) => h(InitiativeTableRowConditions, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('note', {
      header: t('components.encounterTable.headers.note'),
      cell: ({ row }) => h(InitiativeTableRowNotes, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('deathSaves', {
      header: t('components.encounterTable.headers.deathSaves'),
      cell: ({ row }) => h(InitiativeTableRowDeathSaves, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.accessor('concentration', {
      header: t('components.encounterTable.headers.concentration'),
      cell: ({ row }) => h(InitiativeTableRowConcentration, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
    columnHelper.display({
      enableGlobalFilter: false,
      enableSorting: false,
      id: 'actions',
      cell: ({ row }) => h(InitiativeTableRowModify, {
        item: row.original,
        sheet: sheet.value,
        update,
      }),
    }),
  ]
}

export function expandedMarkup(row: Row<InitiativeSheetRow>) {
  return actionsTable(row.original)
}

export const initialState: InitialTableState = {}
