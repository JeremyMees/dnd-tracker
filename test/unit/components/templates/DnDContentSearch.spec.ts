import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it, beforeEach, vi } from 'vitest'
import DnDContentSearch from '~/components/templates/DnDContentSearch.vue'
import { sheet } from '~~/test/unit/fixtures/initiative-sheet'
import open5eItem from '~~/test/unit/fixtures/open5e-item.json'
import armorListing from '~~/test/unit/fixtures/armor-listing.json'

interface Props {
  variant?: 'secondary' | 'background'
  sheet?: InitiativeSheet
  update?: (payload: Omit<Partial<InitiativeSheet>, NotUpdatable | 'campaign'>) => Promise<void>
  allowPin?: boolean
}

const mockUpdate = vi.fn()
const mockToast = vi.fn()

const mockItem = open5eItem as unknown as Open5eItem
const status = ref('success')
const data = ref({ items: armorListing, pages: 1 })

vi.mock('~/components/ui/toast/use-toast', () => ({
  useToast: () => ({ toast: mockToast }),
}))

vi.mock('~~/queries/open5e', () => ({
  useOpen5eListing: () => ({ data, status }),
}))

const props: Props = {
  allowPin: false,
  variant: 'secondary',
  update: mockUpdate,
  sheet: {
    ...sheet,
    info_cards: [mockItem],
  },
}

describe('DnDContentSearch', async () => {
  beforeEach(() => {
    status.value = 'success'
    data.value = { items: armorListing, pages: 1 }
    mockUpdate.mockClear()
    mockToast.mockClear()
  })

  it('Should match snapshot', async () => {
    const component = await mountSuspended(DnDContentSearch, { props })

    expect(component.html()).toMatchSnapshot()
  })

  it('Should render with default props', async () => {
    const component = await mountSuspended(DnDContentSearch, { props })

    expect(component.find('[data-test-search]').exists()).toBeTruthy()
    expect(component.find('[data-test-type]').exists()).toBeTruthy()
    expect(component.find('[data-test-pin-toggle]').exists()).toBeTruthy()
    expect(component.find('[data-test-remove-pins]').exists()).toBeTruthy()
  })

  it('Should not show pin controls when sheet has no info cards', async () => {
    const component = await mountSuspended(DnDContentSearch, {
      props: {
        ...props,
        sheet: {
          ...sheet,
          info_cards: [],
        },
      },
    })

    expect(component.find('[data-test-pin-toggle]').exists()).toBeFalsy()
    expect(component.find('[data-test-remove-pins]').exists()).toBeFalsy()
  })

  it('Should show all items', async () => {
    const component = await mountSuspended(DnDContentSearch, { props })

    const grid = component.find('[data-test-content-grid]')

    expect(grid.exists()).toBeTruthy()
    expect(grid.findAllComponents({ name: 'ContentCard' })).toHaveLength(18)
  })

  it('Should show only pinned items', async () => {
    const component = await mountSuspended(DnDContentSearch, { props })

    await component.find('[data-test-pin-toggle]').trigger('click')

    const grid = component.find('[data-test-content-grid]')

    expect(grid.exists()).toBeTruthy()
    expect(grid.findAllComponents({ name: 'ContentCard' })).toHaveLength(1)
  })

  it('Should handle remove pins', async () => {
    const component = await mountSuspended(DnDContentSearch, { props })

    await component.find('[data-test-remove-pins]').trigger('click')

    expect(mockUpdate).toHaveBeenCalledWith({
      info_cards: [],
    })
  })

  it('Should show toast when trying to pin more than 10 items', async () => {
    const component = await mountSuspended(DnDContentSearch, {
      props: {
        ...props,
        allowPin: true,
        sheet: {
          ...sheet,
          info_cards: Array(10).fill(open5eItem),
        },
      },
    })

    const contentCard = component.findComponent({ name: 'ContentCard' })
    await contentCard.vm.$emit('pin', open5eItem)

    expect(mockToast).toHaveBeenCalledWith({
      title: expect.stringMatching('components.dndContentSearch.toast.maxTitle'),
      description: expect.stringMatching('components.dndContentSearch.toast.maxText'),
      variant: 'destructive',
    })
  })

  it('Should show loading state', async () => {
    status.value = 'pending'

    const component = await mountSuspended(DnDContentSearch, { props })

    expect(component.find('[data-test-loading]').exists()).toBeTruthy()
    expect(component.find('[data-test-not-found]').exists()).toBeFalsy()
    expect(component.find('[data-test-error]').exists()).toBeFalsy()
  })

  it('Should show error state', async () => {
    status.value = 'error'

    const component = await mountSuspended(DnDContentSearch, { props })

    expect(component.text()).toContain('components.dndContentSearch.error')
    expect(component.find('[data-test-error]').exists()).toBeTruthy()
    expect(component.find('[data-test-loading]').exists()).toBeFalsy()
    expect(component.find('[data-test-not-found]').exists()).toBeFalsy()
  })

  it('Should show pagination when there are more than 1 page', async () => {
    data.value = { items: armorListing, pages: 2 }

    const component = await mountSuspended(DnDContentSearch, { props })

    expect(component.find('[data-test-pagination]').exists()).toBeTruthy()
  })

  it('Should not show pagination when there are no pages', async () => {
    data.value = { items: armorListing, pages: 0 }

    const component = await mountSuspended(DnDContentSearch, { props })

    expect(component.find('[data-test-pagination]').exists()).toBeFalsy()
  })
})
