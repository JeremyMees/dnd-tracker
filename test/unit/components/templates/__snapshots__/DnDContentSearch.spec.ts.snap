// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DnDContentSearch > Should match snapshot 1`] = `
"<div class="max-h-full flex flex-col gap-4">
  <div class="flex flex-col gap-2">
    <div class="flex flex-col sm:flex-row items-center gap-x-4 gap-y-2">
      <div id="formkit-stub" modelvalue="" data-test-search="" disabled="false" type="search" label="components.inputs.nameLabel" outer-class="$reset !pb-0 w-full sm:w-auto sm:flex-1"></div>
      <div id="formkit-stub" modelvalue="spells" data-test-type="" disabled="false" type="select" label="components.inputs.typeLabel" options="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" outer-class="$reset !pb-0 w-full sm:w-auto sm:flex-1"></div>
    </div>
    <div class="overflow-hidden">
      <div class="flex gap-2"><button data-test-pin-toggle="" aria-label="components.dndContentSearch.show" class="btn-ghost text-muted-foreground flex items-center gap-x-2"><span class="iconify i-tabler:pin size-4 min-w-4 text-foreground" aria-hidden="true"></span> components.dndContentSearch.show</button><button data-test-remove-pins="" aria-label="components.dndContentSearch.remove" class="btn-ghost flex items-center gap-x-2 text-muted-foreground"><span class="iconify i-tabler:trash size-4 min-w-4 text-foreground" aria-hidden="true"></span> components.dndContentSearch.remove</button></div>
    </div>
  </div>
  <div class="overflow-y-auto">
    <div class="grid gap-4 overflow-y-auto" style="grid-template-columns: repeat(2, minmax(0, 1fr));" data-test-content-grid="">
      <div class="flex flex-col gap-4">
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="el">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Breastplate</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Chain mail</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Half plate</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Leather</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Padded</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Ring mail</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Shield</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Studded Leather</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Unarmored Defense (Barbarian)</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
      </div>
      <div class="flex flex-col gap-4">
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="el">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Chain Shirt</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Draconic Resilience</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Hide</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Mage Armor</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Plate</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Scale mail</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Splint</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Unarmored</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
        <div class="rounded-lg text-card-foreground shadow-sm bg-secondary/50 border-secondary relative border-4" id="">
          <div class="flex flex-col gap-y-1.5 p-6">
            <!--v-if-->
            <h3 class="text-2xl font-semibold leading-none tracking-tight" data-test-title="">Unarmored Defense (Monk)</h3>
          </div>
          <div class="p-6 pt-0">
            <!--v-if-->
            <!--v-if-->
          </div>
          <div class="flex items-center p-6 pt-0">
            <div class="flex justify-end w-full"><button class="flex gap-2 text-foreground" aria-label="actions.readMore">
                <p>actions.readMore</p><span class="iconify i-tabler:chevron-down duration-200 size-6 stroke-2" aria-hidden="true"></span>
              </button></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>"
`;
